parsons/mailchimp/mailchimp.py

```diff
@@ -0,0 +1,175 @@
+                  'since_create_time': since_create_time,
+                  'list_id': list_id,
+                  'folder_id': folder_id,
+                  'member_id': member_id,
+                  'sort_field': sort_field,
+                  'sort_dir': sort_dir}
+
+        response = self.get_request('campaigns', params=params)
+        tbl = Table(response['campaigns'])
+        logger.info(f'Found {tbl.num_rows} campaigns.')
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return None
+
+    def get_members(self, list_id, fields=None,
+                    exclude_fields=None, count=None, offset=None,
+                    email_type=None, status=None, since_timestamp_opt=None,
+                    before_timestamp_opt=None, since_last_changed=None,
+                    before_last_changed=None, unique_email_id=None,
+                    vip_only=False, interest_category_id=None,
+                    interest_ids=None, interest_match=None, sort_field=None,
+                    sort_dir=None, since_last_campaign=None,
+                    unsubscribed_since=None):
+
+        params = {'fields': fields,
+                  'exclude_fields': exclude_fields,
+                  'count': count,
+                  'offset': offset,
+                  'email_type': email_type,
+                  'status': status,
+                  'since_timestamp_opt': since_timestamp_opt,
+                  'before_timestamp_opt': before_timestamp_opt,
+                  'since_last_changed': since_last_changed,
+                  'before_last_changed': before_last_changed,
+                  'unqiue_email_id': unique_email_id,
+                  'vip_only': vip_only,
+                  'interest_category_id': interest_category_id,
+                  'interest_ids': interest_ids,
+                  'interest_match': interest_match,
+                  'sort_field': sort_field,
+                  'sort_dir': sort_dir,
+                  'since_last_campaign': since_last_campaign,
+                  'unsubscribed_since': unsubscribed_since}
+
+        response = self.get_request('lists/' + list_id + '/members',
+                                    params=params)
+        tbl = Table(response['members'])
+        logger.info(f'Found {tbl.num_rows} members.')
+        if tbl.num_rows > 0:
+            return tbl
+        else:
+            return None
+
+    def get_campaign_emails(self, campaign_id, fields=None,
+                            exclude_fields=None, count=None, offset=None,
+                            since=None):
+
+        params = {'fields': fields,
+                  'exclude_fields': exclude_fields,
+                  'count': count,
+                  'offset': offset,
+                  'since': since}
+
+        response = self.get_request('reports/' + campaign_id +
+                                    '/email-activity',
+                                    params=params)
```

Comment:
Consider f-string. See above for example.

Reply:
Solved in 835df40.